parser/parse_test.go

```diff
@@ -253,6 +274,62 @@ const FooEnumSymbol proto.Enumeration = proto.EnumSymbol
 	if p := u.Package("base/foo/proto"); !p.HasImport("base/common/proto") {
 		t.Errorf("Unexpected lack of import line: %#v", p.Imports)
 	}
+	if len(u.Package("base/foo/proto").Constants) != 8 {
+		t.Errorf("Wanted 8 constants, got: %s", spew.Sdump(u.Package("base/foo/proto").Constants))
+	}
+
+	strPtr := func(s string) *string { return &s }
+
+	expectedConst := map[string]*types.Type{
+		"EnumSymbol": &types.Type{
+			Name:       types.Name{Package: "base/foo/proto", Name: "EnumSymbol"},
+			Kind:       types.DeclarationOf,
+			ConstValue: strPtr(`"enumSymbolValue"`),
+		},
+		"FirstDegree": &types.Type{
+			Name:       types.Name{Package: "base/foo/proto", Name: "FirstDegree"},
+			Kind:       types.DeclarationOf,
+			ConstValue: strPtr("0"),
+		},
+		"SecondDegree": &types.Type{
+			Name:       types.Name{Package: "base/foo/proto", Name: "SecondDegree"},
+			Kind:       types.DeclarationOf,
+			ConstValue: strPtr("1"),
+		},
+		"ThirdDegree": &types.Type{
+			Name:       types.Name{Package: "base/foo/proto", Name: "ThirdDegree"},
+			Kind:       types.DeclarationOf,
+			ConstValue: strPtr("2"),
+		},
+		"ConstNineNine": &types.Type{
+			Name:       types.Name{Package: "base/foo/proto", Name: "ConstNineNine"},
+			Kind:       types.DeclarationOf,
+			ConstValue: strPtr("99"),
+		},
+		"ConstHundred": &types.Type{
+			Name:       types.Name{Package: "base/foo/proto", Name: "ConstHundred"},
+			Kind:       types.DeclarationOf,
+			ConstValue: strPtr("100"),
+		},
+		"ConstFloat": &types.Type{
+			Name:       types.Name{Package: "base/foo/proto", Name: "ConstFloat"},
+			Kind:       types.DeclarationOf,
+			ConstValue: strPtr("8782019273372467/1125899906842624"),
```

Comment:
Honestly this seems like a bug in go. That actually works out to 7.79999999999999982236431605997495353221893310546875 and only gets the right answer if you truncate to float64.

https://play.golang.org/p/N6b9whhPoxa

Reply:
It's pretty close though! 😂